src/components/Map/MapLegend.tsx

```diff
@@ -31,27 +34,29 @@ export type MapMarkerLegendItem = {
 };
 
 export type MapMarkerLegendGroup = {
+  disabled?: boolean;
+  tooltip?: string;
+  title: string;
   type: 'marker';
   items: MapMarkerLegendItem[];
 };
 
 export type MapHighlightLegendItem = {
   label: string;
-  style: MapIconComponentStyle | MapFillComponentStyle;
+  style: MapFillComponentStyle;
 };
 
 export type MapHighlightLegendGroup = {
+  disabled?: boolean;
+  tooltip?: string;
+  title: string;
   type: 'highlight';
   items: MapHighlightLegendItem[];
   setVisible?: (visible: boolean) => void;
   visible: boolean;
 };
 
-export type MapLegendGroup = {
-  disabled?: boolean;
-  tooltip?: string;
-  title: string;
-} & (MapMarkerLegendGroup | MapLayerLegendGroup | MapHighlightLegendGroup);
+export type MapLegendGroup = MapMarkerLegendGroup | MapLayerLegendGroup | MapHighlightLegendGroup;
```

Comment:
Could also add a sub type that `MapMarkerLegendGroup`,  `MapLayerLegendGroup`, and `MapHighlightLegendGroup` all inherit that contains `disabled`, `tooltip`, and `title`